## 변경된 점심시간 문제 : 표본 오차란?

링크 : https://hongsi1991.tistory.com/83  ( 참고 : 히스토그램 vs 막대그래프)

모집단에서 복수의 표본을 무작위로 추출하면 곤란한 문제가 생긴다.    
모집단의 성질이 표본별로 달라져 버리는데 이 성질을 표본오차라 한다.  
이 표본오차를 극복하는 것이 확률론이다.  

__모집단__ : 십만명의 인구중 20%의 흡연자가 있는 집단.

In [ ]:
# -*- coding: utf-8 -*-
"""
Created on Mon Dec  9 11:03:34 2019

@author: wdp
"""

import random
from random import shuffle

a = []
people = int(input('전체 인구수를 입력하세요.'))

for i in range(int(people*0.2)): #흡연자를 1로가정
    a.append(1)

for i in range(int(people*0.8)): # 비흡연자를 0으로 가정
    a.append(0)


shuffle(a)


size = int(input('100명의 표본을 몇번 추출할것인지 입력하시오!'))

for i in range(1,size+1):
    b = random.sample(a,100)
    c = (sum(b)/100) * 100

    print('포본', i, '의 흡연률 :', c, '%')

## 예제133. 텍스트 파일을 읽고 출력하기

텍스트 파일을 읽고 그 내용을 화면에 출력하고자 하면 제일 먼저 텍스트 읽기 모드로 파일을 엽니다.  
텍스트 파일을 오픈하면 텍스트 파일을 읽어 내용을 화면에 출력하면 됩니다.  

In [ ]:
f = open("d:\\data\stockcode.txt", "r")
data = f.read() #stockcode.txt 파일의 모든 내용을 한꺼번에 읽는다.
print(data)
f.close() # 파일에 대한 처리가 모두 끝나면 파일 객체의 close() 메소드를 이용해 파일을 닫는다.

## 신경망을 구현할 때 주변 코드로 작성해야 하는 함수

1. __image_load__ : 사진을 신경망에 로드하는 함수 ( 정상폐, 폐결절)
2. __lable_load__ : 훈련 이미지의 라벨을 one hot encoding하는 함수

## 문제233. xray_label_100.txt를 파이썬에서 open해서 열고 print하시오!

In [ ]:
f = open("d:\\data\\xray_label_100.txt", "r")
data = f.read()
print(data)
f.close()

## 문제234. xray_label_100.txt를 읽어서 프린트하는 함수를 생성하시오!

In [ ]:
"""
path = "d:\\data\\xray_label_100.txt"
print(xray_label(path))
"""

def xray_label(path):
    f = open(path, "r")
    data = f.read()
    return data

path = "d:\\data\\xray_label_100.txt"
print(xray_label(path))

## 문제235.   
## 위의 함수를 수정해서 아래와 같이 결과가 출력되게하시오!
## normal은 숫자 0으로 출력하고 patient는 숫자1를 출력되게 하시오!

In [19]:
def label_load(path):
    f = open(path, "r")
    data = f.read()
    L = data.split('\n')
    for i in range(len(L)):
        if L[i] == 'normal':
            L[i] = 0
        elif L[i] == 'patient':
            L[i] = 1
    return L

path = "d:\\data\\xray_label_100.txt"
print(label_load(path))

[0, 1, 1, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 1, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 1, 1, 1, 1, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1]


In [ ]:
def label_load(path):
    f = open(path, "r")
    data = f.read()
    L = data.split('\n')
    for i in range(len(L)):
        if L[i] == 'normal':
            print (0)
        elif L[i] == 'patient':
            print (1)

path = "d:\\data\\xray_label_100.txt"
print(label_load(path))

## 문제236. 아래의 결과를 출력하시오!

In [24]:
# [0 0 0 1 0 0 0 0 0 0]  one hot encoding

import numpy as np

print(np.eye(10)[4]) # 10 : 전체 10자리,  [4] : 4번째 자리를 0으로 만들어라.

[0. 0. 0. 0. 1. 0. 0. 0. 0. 0.]


## 문제237. 아래와 같이 결과를 출력하시오! (폐결절 리스트가 출력되게 하시오)

[0 1] : 폐결절  
[1 0] : 정상폐

In [26]:
import numpy as np

print(np.eye(2)[1])
print(np.eye(2)[0])

[0. 1.]
[1. 0.]


## 문제238. xray_label 함수의 결과가 아래와 같이 one hot encoding된 결과로 출력되게하시오!

In [ ]:
"""
print(xray_label(path))

normal -> [1 0]
patient --> [0 1]

"""

def label_load(path):
    f = open(path, "r")
    data = f.read()
    L = data.split('\n')
    for i in range(len(L)):
        if L[i] == 'normal':
            print(np.eye(2)[0])
        elif L[i] == 'patient':
            print(np.eye(2)[1])
            


path = "d:\\data\\xray_label_100.txt"
label_load(path)

In [ ]:
import numpy as np
def xray_label(path) : 

    f = open(path,'r')

    data = f.read().split('\n')

    arr = []

    

    for i in data :

        if i == 'patient' :

            arr.append(list(map(int,np.eye(2)[1])))

        else :

            arr.append(list(map(int,np.eye(2)[0])))



    return arr



path = 'd:\\data\\xray_label_100.txt'

print(xray_label(path))

## 예제134. 텍스트 파일을 한 줄씩 읽고 출력하기(readline)

텍스트 파일의 용량이 매우 클 경우에는 read()로 한꺼번에 파일의 내용을 읽어들이는 것은 메모리 문제를 야기할 수 있습니다.  
이 경우 텍스트 파일 내용을 한 줄 단위로 읽고 작업을 수행하면 됩니다.  
readline()은 텍스트 파일에서 한 줄을 읽습니다.  
한 줄을 읽고나면 파일을 읽기 시작하는 위치는 그 다음줄의 맨 처음이 됩니다.  

In [37]:
f = open("d:\\data\stockcode.txt", "r")
line = f.readline()
print(line)

000020 동화약품



## 문제239. 위의 코드에 루프문을 써서 텍스트 파일의 내용이 전부 다 출력되게 하시오!

In [81]:
f = open("d:\\data\stockcode.txt", "r")
line = f.readline() # 000020 동화약품

while line:
    print(line, end = '') # 처음 print : 000020 동화약품 & # end = '' --> print사이의 공간을 원래는 \n인데 ''(null)로 설정하겠다는 뜻
    line = f.readline()  # 000040 S&T모터스이 line에 저장된 상태.
    
f.close()

000020 동화약품
000040 S&T모터스
000050 경방
000060 메리츠화재
000070 삼양사
000071 삼양사우
000100 유한양행
000101 유한양행우
000120 대한통운
000140 하이트홀딩스
000141 하이트홀딩스우
000150 두산
000151 두산우
000152 두산2우B
000180 성창기업지주
000210 대림산업
000211 대림산업우
000220 유유제약
000221 유유제약1우
000222 유유제약2우B
000230 일동제약
000240 한국타이어
000270 기아차
000300 대유신소재
000320 디피아이홀딩스
000321 디피아이홀딩스우
000322 디피아이홀딩스2B
000360 삼환기업
000361 삼환기업우
000370 한화손해보험
000390 삼화페인트
000400 롯데손해보험
000420 로케트전기
000421 로케트전우
000430 대원강업
000470 그린손해보험
000480 조선내화
000490 대동공업
000500 가온전선
000520 삼일제약
000540 흥국쌍용화재
000541 흥국쌍용화재우
000542 흥국쌍용화재2우B
000590 조선선재
000610 제일화재
000640 동아제약
000650 천일고속
000660 하이닉스
000670 영풍
000680 LS네트웍스
000681 LS네트웍스우
000700 한진해운
000720 현대건설
000721 현대건설우
000760 이화산업
000790 C&상선
000800 경남기업
000810 삼성화재
000811 삼성화재우
000830 삼성물산
000831 삼성물산우
000850 화천기공
000860 건설화학
000880 한화
000881 한화우
000890 보해양조
000891 보해양조우
000910 유니온
000950 전방
000970 한국주철관
000990 동부하이텍
000991 동부하이텍1우
000992 동부하이텍2우B
001020 페이퍼코리아
001040 CJ
001041 CJ우
001042 CJ2우B
001043 CJ3우B
001060 중

In [78]:
# for문 버전

f = open("d:\\data\stockcode.txt", "r")
line = f.readline()

for i in f.readlines(): # readlines로 전체를 읽어온다.
    print(i, end = '') # print사이의 공간을 원래는 \n인데 ''(null)로 설정하겠다는 뜻

    
f.close()

000040 S&T모터스


## 예제135. 화면에서 사용자 입력을 받고 파일로 쓰기(write)

사용자로부터 입력받은 텍스트를 파일로 저장하려면 파일을 텍스트 쓰기 모드로 오픈하고   
파일 객체의 write()를 이용해 데이터 파일에 기록합니다.

In [ ]:
text = input('파일에 저장할 내용을 입력하세요~')
f = open("d:\\data\\mydata.txt", "w")
f.write(text)
f.close()

## 문제240.   
## 금요일에 냈던 마지막 문제인 중심극한의 정리의 결과를 d드라이브 밑에  
## mydata2.txt라는 이름으로 생성하시오!  

In [63]:
def dice_average(cnt):
    
    import random
    import numpy as np

    dice = 100000

    c = []

    for i in range(cnt):

        a = [1,2,3,4,5,6] * dice

        b = random.sample(a, 10)

        c.append(np.mean(b))
        d = np.mean(c)

    return d

cnt = [10,50,100,500,1000]

result = map(dice_average, cnt)

f = open("d:\\data\\mydata2.txt", "w")

f.write(str(list(result))) #결과값이 map이라서 앞에 list로 감싸준다.

f.close()

## 예제137. 텍스트 파일에 한 줄씩 쓰기 (writelines)

파일 객체의 writelines()는 텍스트 문자열이나 텍스트 문자열이 요소로 되어있는 리스트를 인자로 받아  
파일에 한 줄씩 기록합니다.  
리스트가 인자인 경우 writelines()는 리스트의 요소를 하나의 문자열로 결합하여 파일에 한꺼번에 기록합니다.

★ writelines 는 리스트 자료형도 파일에 저장할 수 있다.

In [58]:
listdata = ['a', 'b', 'c', 'd', 'e']
result1 = sorted(listdata)

print(result1)

f = open("d:\\data\\mydata4.txt", "w")
f.writelines(result1) #리스트 안의 요소가 문자일시 write를 하면 오류가남
f.close()

[1, 2, 2, 3, 5, 7, 8]


### ※ 위와 같이 리스트 안의 요소가 숫자가 아니라 문자이면 write로 하면 에러가 나고 writelines로 하면 수행된다.

## 예제138. 텍스트 파일 복사하기(read,write)

텍스트 파일을 복사하는 방법은 매우 간단하다.  
일반적으로 텍스트 파일의 크기는 대용량이 아니므로 원본 텍스트 파일의 모든 내용을 read()로 읽고 다른 파일이름으로    
텍스트 쓰기 모드로 오픈한후 write()로 읽은 내용을 기록하면 된다.  

In [ ]:
f = open("d:\\data\\stockcode.txt", "r")
h = open("d:\\data\\stockcode_copy.txt", "w")

data = f.read()
h.write(data)

f.close()
h.close()

## 예제139. 바이너리 파일 복사하기(read, write)

일반적으로 바이너리 파일은 용량이 큰 경우가 많다.  
예를들어, 고화질 영화를 담은 동영상 파일의 경우 10GB가 넘기도 한다.  
대용량 파일의 내용을 한 번에 읽고 쓰는 것은 메모리 용량의 한계로 불가능할 때가 있고 성능적인 면에서도 효율적이지 못하다.  
이러한 경우 파일을 처음부터 끝까지 일정한 크기 단위로 읽고 쓰면 메모리 문제를 해결할 수 있다.

In [66]:
bufsize = 1024 # 1kb = 1024byte

f = open("d:\\data\\lena.png", "rb")
h = open("d:\\data\\lena_copy.png", "wb")

data = f.read(bufsize) # 1kb씩 읽어오는 상황

while data:
    h.write(data)
    data = f.read(bufsize) # 파일이 너무 크기때문에 while반복문을 이용하여 1kb씩 읽고 복사하는(쓰는) 상황.
    
f.close()
h.close()

In [67]:
bufsize = 1024*5 # 1kb = 1024byte

f = open("d:\\data\\lena.png", "rb")
h = open("d:\\data\\lena_copy2.png", "wb")

data = f.read(bufsize) # 1kb씩 읽어오는 상황
while data:
    h.write(data)
    data = f.read(bufsize) # 파일이 너무 크기때문에 while반복문을 이용하여 1kb씩 읽고 복사하는(쓰는) 상황.
f.close()
h.close()

## 예제140. 파일을 열고 자동으로 닫기(with ~ as)

with ~ as 절을 사용하게 되면 f.close()를 명시하지 않아도 되므로  
프로그래머가 실수로 f.close()를 작성하지 않아서 코드를 실행할 때 메모리 부족 오류가 나는것을 예방 할 수 있다.

In [ ]:
# 1. with ~ as 절을 사용 안했을 때

f = open("d:\\data\\stockcode.txt", "r")
data = f.readlines()
print(data)
f.close()

In [84]:
# 2. with ~as 절을 사용했을 때

with open("d:\\data\\stockcode.txt", "r") as f:
    for i in f.readlines():
        print(i, end = '')

000020 동화약품
000040 S&T모터스
000050 경방
000060 메리츠화재
000070 삼양사
000071 삼양사우
000100 유한양행
000101 유한양행우
000120 대한통운
000140 하이트홀딩스
000141 하이트홀딩스우
000150 두산
000151 두산우
000152 두산2우B
000180 성창기업지주
000210 대림산업
000211 대림산업우
000220 유유제약
000221 유유제약1우
000222 유유제약2우B
000230 일동제약
000240 한국타이어
000270 기아차
000300 대유신소재
000320 디피아이홀딩스
000321 디피아이홀딩스우
000322 디피아이홀딩스2B
000360 삼환기업
000361 삼환기업우
000370 한화손해보험
000390 삼화페인트
000400 롯데손해보험
000420 로케트전기
000421 로케트전우
000430 대원강업
000470 그린손해보험
000480 조선내화
000490 대동공업
000500 가온전선
000520 삼일제약
000540 흥국쌍용화재
000541 흥국쌍용화재우
000542 흥국쌍용화재2우B
000590 조선선재
000610 제일화재
000640 동아제약
000650 천일고속
000660 하이닉스
000670 영풍
000680 LS네트웍스
000681 LS네트웍스우
000700 한진해운
000720 현대건설
000721 현대건설우
000760 이화산업
000790 C&상선
000800 경남기업
000810 삼성화재
000811 삼성화재우
000830 삼성물산
000831 삼성물산우
000850 화천기공
000860 건설화학
000880 한화
000881 한화우
000890 보해양조
000891 보해양조우
000910 유니온
000950 전방
000970 한국주철관
000990 동부하이텍
000991 동부하이텍1우
000992 동부하이텍2우B
001020 페이퍼코리아
001040 CJ
001041 CJ우
001042 CJ2우B
001043 CJ3우B
001060 중

## 예제141. mySQL과 파이썬 연동하기!

Data를 저장하고 관리하는 Database?

1. 오라클
2. mysql
3. mssql
4. postgresql

## 예제142. HTML 기본 문법

### html 이란 ? 

### Hyper Text Markup Language 의 약자이고 여러개의 테그(tag) 를 연결해서 모아놓은 문서  

## ※ 웹스크롤링

### 우리가 원하는 정보(데이터)를 가져오기 위한 일종의 수단

## 문제241. 간단한 html 문서를 만들어 보시오!

메모장을 열고 아래와 같이 코딩을 하세요~

다른이름으로 저장하는데 a.html로 저장하시오!

In [ ]:
<html><head><title> 김인엽님의 오늘 일정 </title></head>
<body>
<p class = "title"> 김인엽님은 오늘 점심 명동 칼국수를 먹었습니다</p>
</body>
</html>

## 문제242. 위의 글씨를 진하게 하시오~ < b >

In [ ]:
<html><head><title> 김인엽님의 오늘 일정 </title></head>
<body>
<p class = "title"> <b> 김인엽님은 오늘 점심 명동 칼국수를 먹었습니다 </b> </p>
</body>
</html>

## 문제243. 위의 글씨에 밑줄을 그어보시오~ < u >

In [ ]:
<html><head><title> 김인엽님의 오늘 일정 </title></head>
<body>
<p class = "title"> <b><u> 김인엽님은 오늘 점심 명동 칼국수를 먹었습니다 </u></b> </p>
</body>
</html>

## 문제244. 위의 글씨체를 이탤릭체로 변경하시오! < i >

In [ ]:
<html><head><title> 김인엽님의 오늘 일정 </title></head>
<body>
<p class = "title"> <b><u><i> 김인엽님은 오늘 점심 명동 칼국수를 먹었습니다 </i></u></b> </p>
</body>
</html>

## 문제245. p tage를 추가해서 제목과 내용을 나누시오.

In [ ]:
<html><head><title> 김인엽님의 오늘 일정 </title></head>
<body>
<p class = "title"> <b><u><i> 김인엽님의 점심 </i></u></b> </p>
<p class = "content"> 인엽이는 성민이와 명동 칼국수에 만두를 같이 먹었습니다. </p>
</body>
</html>


<html>
<head>
<title> 김인엽님의 오늘 일정 </title>
</head>
<body>
<p class = "title"> <b><u><i> 김인엽님의 점심 </i></u></b> </p>
<p class = "content"> 인엽이는 성민이와 명동 칼국수에 만두를 같이 먹었습니다. </p>
</body>
</html>


__윈도우 개발자모드 --> F12__ , 여기에서 html이 어떻게 쓰여졌는지 확인할 수 있다.

## 문제246. 위의 html문서의 본문에 링크를 걸어보시오

In [ ]:
<html><head><title> 김인엽님의 오늘 일정 </title></head>
<body>

<p class = "title"> <b><u><i> 김인엽님의 점심 </i></u></b> </p>
<p class = "content"> 인엽이는 성민이와 명동 칼국수에 만두를 같이 먹었습니다. 
<a href = "http:\\cafe.daum.net/oracleoracle"  class = "cafe1"  id = "link1">다음카페</a>
</p>

</body>
</html>

## 문제247. 아래와 같이 링크를 만드시오

김인엽님의 점심

인엽이는 성밍이와 명동 칼국수에 만두를 같이 먹었습니다.  
다음카페  
네이버  
구글  

In [ ]:
<html><head><title> 김인엽님의 오늘 일정 </title></head>
<body>

<p class = "title"> <b><u><i> 김인엽님의 점심 </i></u></b> </p>
<p class = "content"> 인엽이는 성민이와 명동 칼국수에 만두를 같이 먹었습니다. <br/>
<a href = "http:\\cafe.daum.net/oracleoracle"  class = "cafe1"  id = "link1">다음카페</a><br/>
<a href = "https://www.naver.com/" class = "cafe1"  id = "link1">네이버</a><br/>
<a href = "http:\\google.com"  class = "cafe1"  id = "link1">구글</a>
</p>

</body>
</html>

## 문제248. 위에서 만든 a.html은 d 드라이브 밑에 가져다 놓고 beautiful soup 모듈을 이용해서  
## a.html문서의 title을 검색하시오.



### ★ beautiful soup ★: 웹 스크롤링을 위하여 최적화된 모듈!!

In [91]:
from bs4 import BeautifulSoup

with open("d:\\data\\a.html", encoding = "UTF-8") as a:
    soup = BeautifulSoup(a, "lxml") # 이렇게 하면 soup에 html 전체가 soup에 담긴다.
    
print(soup)
print(soup.title)

<html><head><title> 김인엽님의 오늘 일정 </title></head>
<body>
<p class="title"> <b><u><i> 김인엽님의 점심 </i></u></b> </p>
<p class="content"> 인엽이는 성민이와 명동 칼국수에 만두를 같이 먹었습니다. <br/>
<a class="cafe1" href="http:\\cafe.daum.net/oracleoracle" id="link1">다음카페</a><br/>
<a class="cafe1" href="https://www.naver.com/" id="link1">네이버</a><br/>
<a class="cafe1" href="http:\\google.com" id="link1">구글</a>
</p>
</body>
</html>
<title> 김인엽님의 오늘 일정 </title>


## 문제249. 위에서 긁어온 title에서 text만 검색되게 하시오.

In [93]:
from bs4 import BeautifulSoup

with open("d:\\data\\a.html", encoding = "UTF-8") as a:
    soup = BeautifulSoup(a, "lxml") # 이렇게 하면 soup에 html 전체가 soup에 담긴다.

print(soup.title.string) #뒤에 .string를 붙여준다.

 김인엽님의 오늘 일정 


## 문제250. a.html 문서에서 p태그에 대한 html을 검색하시오!

In [96]:
from bs4 import BeautifulSoup

with open("d:\\data\\a.html", encoding = "UTF-8") as a:
    soup = BeautifulSoup(a, "lxml") # 이렇게 하면 soup에 html 전체가 soup에 담긴다.

print(soup.find('p')) # find를 이용해서 p태그를 가져올 수 있다.

<p class="title"> <b><u><i> 김인엽님의 점심 </i></u></b> </p>


## 문제251. a.html 문서에서 p 태그를 모두 검색하시오!

In [97]:
from bs4 import BeautifulSoup

with open("d:\\data\\a.html", encoding = "UTF-8") as a:
    soup = BeautifulSoup(a, "lxml") # 이렇게 하면 soup에 html 전체가 soup에 담긴다.

print(soup.find_all('p')) # find를 이용해서 p태그를 가져올 수 있다.

[<p class="title"> <b><u><i> 김인엽님의 점심 </i></u></b> </p>, <p class="content"> 인엽이는 성민이와 명동 칼국수에 만두를 같이 먹었습니다. <br/>
<a class="cafe1" href="http:\\cafe.daum.net/oracleoracle" id="link1">다음카페</a><br/>
<a class="cafe1" href="https://www.naver.com/" id="link1">네이버</a><br/>
<a class="cafe1" href="http:\\google.com" id="link1">구글</a>
</p>]


## 문제252. a.html문서에서 a 태그를 모두 검색하시오!

In [98]:
from bs4 import BeautifulSoup

with open("d:\\data\\a.html", encoding = "UTF-8") as a:
    soup = BeautifulSoup(a, "lxml") # 이렇게 하면 soup에 html 전체가 soup에 담긴다.

print(soup.find_all('a')) # find를 이용해서 p태그를 가져올 수 있다.

[<a class="cafe1" href="http:\\cafe.daum.net/oracleoracle" id="link1">다음카페</a>, <a class="cafe1" href="https://www.naver.com/" id="link1">네이버</a>, <a class="cafe1" href="http:\\google.com" id="link1">구글</a>]


## 문제253. a.html 문서에서 a태그의 href 링크의 url만 긁어 오시오!

In [100]:
from bs4 import BeautifulSoup

with open("d:\\data\\a.html", encoding = "UTF-8") as a:
    soup = BeautifulSoup(a, "lxml") # 이렇게 하면 soup에 html 전체가 soup에 담긴다.

for i in soup.find_all('a'):
    print(i) # 리스트 형이기 때문에 하나하나씩 for문으로 가져와야된다.
    print(i.get("href")) # i.get("href")로 url만 깔끔하게 긁어올 수 있다.

<a class="cafe1" href="http:\\cafe.daum.net/oracleoracle" id="link1">다음카페</a>
http:\\cafe.daum.net/oracleoracle
<a class="cafe1" href="https://www.naver.com/" id="link1">네이버</a>
https://www.naver.com/
<a class="cafe1" href="http:\\google.com" id="link1">구글</a>
http:\\google.com


## 문제254. a.html 문서에서 html 코드 말고 text만 출력하시오!

In [108]:
from bs4 import BeautifulSoup

with open("d:\\data\\a.html", encoding = "UTF-8") as a:
    soup = BeautifulSoup(a, "lxml") # 이렇게 하면 soup에 html 전체가 soup에 담긴다.
    
print(soup.get_text())

김인엽님의 오늘 일정

 김인엽님의 점심 
 인엽이는 성민이와 명동 칼국수에 만두를 같이 먹었습니다.
 다음카페
네이버
구글





## 문제255. 위의 텍스트를 한줄로 나오게 하시오!

In [109]:
from bs4 import BeautifulSoup

with open("d:\\data\\a.html", encoding = "UTF-8") as a:
    soup = BeautifulSoup(a, "lxml") # 이렇게 하면 soup에 html 전체가 soup에 담긴다.
    
print(soup.get_text(" ", strip = True)) # strip 잘라내다. 즉, 공백을 잘라버려라!b

김인엽님의 오늘 일정 김인엽님의 점심 인엽이는 성민이와 명동 칼국수에 만두를 같이 먹었습니다. 다음카페 네이버 구글


## 문제256. (오늘의 마지막 문제) mySQL 설치후 EMP만들고 select 까지 하세요~